test/BaseTestEntityWithSources.cpp

```diff
@@ -82,4 +82,30 @@ void BaseTestEntityWithSources::testSourceVectorSetter() {
     std::vector<Source> deleter;
     da.sources(deleter);
     CPPUNIT_ASSERT(da.sourceCount() == 0);
-}
\ No newline at end of file
+}
+
+void BaseTestEntityWithSources::testSourceOrder() {
+    std::vector<std::string> names = { "source_a", "source_b", "source_c", "source_d", "source_e" };
+    std::vector<Source> sources;
+    Group append_group = block.createGroup("ag", "test");
+    Group set_group = block.createGroup("sg", "test");
+
+    for (auto it = names.begin(); it != names.end(); it++) {
```

Comment:
For future reference, "range based for loops" are nicer; i.e. `for (const auto &name : names) { ... }`.
I assume this was c&p'ed from code above, so one could make an argument about consistency, even though the iterator based approach is just so C++03 ;). OTOH you are using range based loop below, sooo ;) .. Anyway, I'll allow it ;)